lib/phoenix/live_dashboard/system_info.ex

```diff
@@ -235,10 +240,75 @@ defmodule Phoenix.LiveDashboard.SystemInfo do
     end
   end
 
+  ## Socket callbacks
+
+  def sockets_callback(search, sort_by, sort_dir, limit) do
+    sorter = if sort_dir == :asc, do: &<=/2, else: &>=/2
+
+    sockets =
+      for port <- Port.list(), info = Port.info(port), show_socket?(info) do
```

Comment:
Please see if you can mirror the other callbacks. The idea is that you will do something like:

```suggestion
      for port <- Port.list(), info = socket_info(port), show_socket?(info, search) do
```

And the `socket_info` will already return the info with all relevant fields.